q2_fmt/_ancombc.py

```diff
@@ -0,0 +1,86 @@
+                            level_delimiter=None):
+    filter = ctx.get_action('feature_table', 'filter_samples')
+    ancombc = ctx.get_action('composition', 'ancombc')
+    da_barplot = ctx.get_action('composition', 'da_barplot')
+    results = []
+
+    _, _ = _check_for_time_column(metadata.to_dataframe(),
+                                  time_column)
+    _ = _check_reference_column(metadata.to_dataframe(),
+                                reference_column)
+
+    ids_to_keep = get_baseline_donor_md(metadata=metadata,
+                                        reference_column=reference_column,
+                                        time_column=time_column,
+                                        baseline_timepoint=baseline_timepoint)
+
+    filtered_table, = filter(table=table,
+                             metadata=Metadata(ids_to_keep))
+    dataloaf = ancombc(table=filtered_table, metadata=Metadata(ids_to_keep),
+                       reference_levels=["type::donor"], formula='type')
+    results += dataloaf
+    viz = da_barplot(data=dataloaf[0], significance_threshold=0.05,
+                     level_delimiter=level_delimiter)
+    results += viz
+    return tuple(results)
+
+
+def get_baseline_donor_md(metadata, reference_column, time_column,
+                          baseline_timepoint):
+    """Checks if column is in the metdata
+    Checks that column is in the metdata and creates helpful error messages.
+    Parameters
+    ----------
+    metadata: pd.Dataframe
+        Study `Metadata`
+    reference_column: str
+       name of reference column in `Metadata` column
+    time_column: str
+       name of reference column in `Metadata` column
+    baseline_timepoint: str
+        timepoint that represents baseline
+    Examples
+    --------
+    >>> metadata = pd.DataFrame({'id': ['sample1', 'sample2', 'donor1'],
+                   'reference': ['donor1', 'donor1', np.nan],
+                   'time': [1, 2, np.nan],
+                   'subject': ['sub1','sub1', np.nan]}).set_index('id')
+    >>> time_column = 'time'
+    >>> reference_column = 'reference'
+    >>>  baseline_timepoint = '1'
+    >>> get_baseline_donor_md(metadata, reference_column, time_column,
+                              baseline_timepoint)
+        pd.DataFrame({'id': ['sample1', 'donor1'],
+                   'reference': ['donor1', np.nan],
+                   'time': [1,np.nan],
+                   'subject': ['sub1',np.nan]}).set_index('id')
+    """
+    md_df = metadata.to_dataframe()
+    ids_to_keep =\
+        pd.Series(index=md_df[reference_column].dropna().unique(),
+                  data='donor', name='type')
+    for tp, tp_df in md_df.groupby(by=time_column):
+        if tp == float(baseline_timepoint):
+            ids_to_keep = pd.concat([ids_to_keep,
+                                     pd.Series(index=tp_df.index.to_list(),
+                                               data='baseline', name=type)])
```

Comment:
You can make pandas do the heavy lifting here for the filtering and iteration

```suggestion
    ids_to_keep = pd.concat([ids_to_keep,  pd.Series(index=md_df[md_df[time_column] == float(baseline_timepoint)].index.to_list(), data='baseline', name=type)])

```